awscommons/v2/s3.go

```diff
@@ -0,0 +1,36 @@
+// NewS3Client will return a new AWS SDK client for interacting with AWS Secrets Manager.
+package awscommons
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+
+	"github.com/gruntwork-io/go-commons/errors"
+)
+
+// UploadObjectString will upload the provided string to the given S3 bucket as an object under the specified key.
+func UploadObjectString(opts *Options, bucket string, key string, contents string) error {
```

Comment:
NIT:

```suggestion
func UploadObjectString(opts *Options, bucket, key, contents string) error {
```

Reply:
For some reason that syntax doesn't stick with me... Have to keep at it until it's wired in my brain I guess. Will update in a follow up PR to avoid a review cycle blocking this.

Reply:
https://github.com/gruntwork-io/go-commons/pull/53